src/serialize.h

```diff
@@ -1091,7 +1092,7 @@ inline void WriteCompactSize(SizeComputer &s, uint64_t nSize)
 }
 
 template <typename T>
-size_t GetSerializeSize(const T& t)
+uint64_t GetSerializeSize(const T& t)
```

Comment:
I'm not sure I understand why we need to propagate this being 64 bits instead of clamping this locally to 32 here, something like:
```C++
template <typename T>
uint32_t GetSerializeSize(const T& t)
{
    const uint64_t size{(SizeComputer() << t).size()};
    return uint32_t(std::clamp<int64_t>(size, 0, std::numeric_limits<uint32_t>::max()));
}
```